File: app/src/main/java/org/michaelbel/moviemade/ui/modules/movie/MovieMvp.java
package org.michaelbel.moviemade.ui.modules.movie;

import com.arellomobile.mvp.MvpView;

import org.michaelbel.moviemade.data.dao.Movie;

public interface MovieMvp extends MvpView {

    void setPoster(String posterPath);

    void setMovieTitle(String title);

    void setOverview(String overview);

    void setVoteAverage(float voteAverage);

    void setVoteCount(int voteCount);

    void setReleaseDate(String releaseDate);

    void setOriginalLanguage(String originalLanguage);

    void setRuntime(String runtime);

    void setTagline(String tagline);

    void setURLs(String imdbId, String homepage);

    void setWatching(boolean watch);

    void showConnectionError();

    void showComplete(Movie movie);
}

File: app/src/main/java/org/michaelbel/moviemade/ui/modules/movie/MovieMvp.kt
package org.michaelbel.moviemade.ui.modules.movie

import com.arellomobile.mvp.MvpView
import org.michaelbel.moviemade.data.entity.Mark
import org.michaelbel.moviemade.data.entity.Movie

interface MovieMvp : MvpView {

    fun setPoster(posterPath: String)

    fun setMovieTitle(title: String)

    fun setOverview(overview: String)

    fun setVoteAverage(voteAverage: Float)

    fun setVoteCount(voteCount: Int)

    fun setReleaseDate(releaseDate: String)

    fun setOriginalLanguage(originalLanguage: String)

    fun setRuntime(runtime: String)

    fun setTagline(tagline: String)

    fun setURLs(imdbId: String, homepage: String)

    fun setStates(fave: Boolean, watch : Boolean)

    fun onFavoriteChanged(mark: Mark)

    fun onWatchListChanged(mark: Mark)

    fun setCredits(casts: String, directors: String, writers: String, producers: String)

    fun setConnectionError()

    fun setGenres(genres : List<Int>)

    fun showComplete(movie: Movie)
}